### PR TITLE
PYTHON-4497 Fix handling of Python executable in run-tests

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: .evergreen/run-tests.sh [hunk 1]
```diff
@@ -31,9 +31,6 @@ set -o xtrace
 AUTH=${AUTH:-noauth}
 SSL=${SSL:-nossl}
 TEST_ARGS="${*:1}"
-PYTHON=$(which python)
-# TODO: Remove when we drop PyPy 3.8 support.
-OLD_PYPY=$(python -c "import sys; print(sys.implementation.name.lower() == 'pypy' and sys.implementation.version < (7, 3, 12))")
```
Comment (ShaneHarvey) [inline .evergreen/run-tests.sh:36]: We can remove the uses of OLD_PYPY now too.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
To clarify, is this fixing a bug or just cleaning up old code?

**Comment by blink1073** [comment]
It what trigged by work in https://github.com/mongodb/winkerberos/pull/92

File: .evergreen/run-tests.sh (hunk 1 above)
Comment (blink1073) [reply .evergreen/run-tests.sh:36]: Done

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**